test/unit_tests/apps/retirement/js/data/index-spec.js

```diff
@@ -0,0 +1,45 @@
+import {
+  benefits,
+  lifetime,
+  fetchApiData,
+  updateDataFromApi,
+} from '../../../../../../cfgov/unprocessed/apps/retirement/js/data';
+import { r1, r2, r3 } from './responses.js';
+import fetchMock from 'jest-fetch-mock';
+
+fetchMock.enableMocks();
+
+describe('fetchApiData', function () {
+  it('fetches data', function () {
+    fetch.mockResponseOnce(JSON.stringify({ data: '12345' }));
+    fetchApiData('11-11-1965', 34567).then((res) => {
+      expect(res.data).toEqual('12345');
+    });
+  });
```

Comment:
Sonarcloud is flagging this. Looks like you need:

```js
it('fetches data', async function () {
    fetch.mockResponseOnce(JSON.stringify({ data: '12345' }));
    await fetchApiData('11-11-1965', 34567).then((res) => {
      expect(res.data).toEqual('12345');
    });
  });
```

Reply:
Fixed in 22f698dc9e1629f4ef613265b3abed74cb2a39eb by explicitly handling the reject case